### PR TITLE
Add more wallet signing tests

### PR DESCRIPTION
**Pull request by cgewecke**
Adds tests for:
+ logic added in #3105 to bubble up errors when using eth_sendTransaction with the Web3 wallet
+ accounts.signTransaction without a tx object
+ accounts.signTransaction when ethereumjs-tx does not validate

(Trying to improve the coverage a bit with these, but also found a small bug - see below)

[EDIT 27/10 - The coveralls-github hook looks broken atm, these jobs are [getting processed there][1] but there's no feedback]

[1]: https://coveralls.io/repos/53203/builds

### CODE REVIEW

File: packages/web3-eth-accounts/src/index.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
         error = false,
         transactionOptions = {},
         result,
-        hasTxSigningOptions = !!((tx.chain && tx.hardfork) || tx.common);
+        hasTxSigningOptions = !!(tx && ((tx.chain && tx.hardfork) || tx.common));
```
Comment (cgewecke) [inline packages/web3-eth-accounts/src/index.js:132]: Fixes small bug introduced in #3143. The check for whether or not tx is defined happens after this list of variables is set.

File: test/e2e.method.signing.js [hunk 2]
```diff
@@ -9,6 +9,14 @@ describe('transaction and message signing [ @E2E ]', function() {
     let web3;
     let accounts;
     let wallet;
+    let basic;
+    let instance;
+
+    var basicOptions = {
```
Comment (nivida) [inline test/e2e.method.signing.js]: could you use let or const?

File: test/e2e.method.signing.js [hunk 3]
```diff
@@ -173,6 +184,89 @@ describe('transaction and message signing [ @E2E ]', function() {
...
+        try {
+            await web3.eth.accounts.signTransaction(undefined, wallet[0].privateKey);
+            assert.fail()
+        } catch (err) {
+            assert(err.message.includes('No transaction object'));
+        }
+    });
+
+    it('transactions sent with wallet error correctly (with receipt)', async function(){
+        var data = instance
```
Comment (nivida) [inline test/e2e.method.signing.js]: ``const`` can be used (in the whole file)

File: test/e2e.method.signing.js [hunk 4]
```diff
@@ -173,6 +184,89 @@ describe('transaction and message signing [ @E2E ]', function() {
...
+    it('accounts.signTransaction errors when no transaction is passed', async function(){
+        try {
+            await web3.eth.accounts.signTransaction(undefined, wallet[0].privateKey);
+            assert.fail()
+        } catch (err) {
+            assert(err.message.includes('No transaction object'));
+        }
+    });
+
+    it('transactions sent with wallet error correctly (with receipt)', async function(){
```
Comment (nivida) [inline test/e2e.method.signing.js]:
```suggestion
    it('transactions sent with wallet throws error correctly (with receipt)', async function(){
```

File: test/e2e.method.signing.js [hunk 5]
```diff
@@ -173,6 +184,89 @@ describe('transaction and message signing [ @E2E ]', function() {
...
+            to: instance.options.address,
+            data: data,
+            gasPrice: '1',
+            gas: 4000000
+        }
+
+        try {
+            await web3.eth.sendTransaction(tx);
+            assert.fail();
+
```
Comment (nivida) [inline test/e2e.method.signing.js]:
```suggestion
```

**Review by nivida — APPROVED** [review]
LGTM - Just added some code style comments

File: test/e2e.method.signing.js (hunk 2 above)
Comment (cgewecke) [reply test/e2e.method.signing.js]: Fixed in 269d11943c0db30f11effb227856ecb473bec07c

File: test/e2e.method.signing.js (hunk 3 above)
Comment (cgewecke) [reply test/e2e.method.signing.js]: Fixed in 269d11943c0db30f11effb227856ecb473bec07c

File: test/e2e.method.signing.js (hunk 5 above)
Comment (cgewecke) [reply test/e2e.method.signing.js]: Fixed in 269d11943c0db30f11effb227856ecb473bec07c

**Pull request merged**